pkg/apis/core/validation/validation.go

```diff
@@ -2711,6 +2711,26 @@ func ValidateVolumeDevices(devices []core.VolumeDevice, volmounts map[string]str
 	return allErrs
 }
 
+func validateResourceClaimNames(claimNames []string, podClaimNames sets.String, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	names := sets.NewString()
+	for i, name := range claimNames {
```

Comment:
I wanted to avoid the allocation during the normal code flow when there are no validation errors and `fldPath.Index(i)` doesn't need to be called. Does that make sense? I can change it if you prefer to avoid the code duplication.

Reply:
up to you